File: src/main/java/net/onelitefeather/stardust/api/user/IUser.java
package net.onelitefeather.stardust.api.user;

import net.kyori.adventure.text.Component;
import org.bukkit.Location;
import org.bukkit.entity.Player;

import java.util.List;
import java.util.UUID;

public interface IUser {

    void setId(int id);

    int getId();

    UUID getUniqueId();

    void setUniqueId(UUID uniqueId);


    String getName();

    void setName(String name);

    void setDisplayName(String displayName);

    String getDisplayName();

    void setFlying(boolean flying);

    boolean isFlying();

    void setVanished(boolean vanished);

    boolean isVanished();

    Player getBase();

    boolean toggleVanish();

    void hidePlayer();

    boolean kick(Component message);

    void checkCanFly();


}

File: src/main/kotlin/net/onelitefeather/stardust/api/IUser.kt
package net.onelitefeather.stardust.api

import net.kyori.adventure.text.Component
import org.bukkit.entity.Player
import java.util.UUID


interface IUser {

    fun getId(): Long

    fun getUniqueId(): UUID

    fun getName(): String

    fun setName(name: String): IUser

    fun setDisplayName(displayName: String)

    fun getDisplayName(): String

    fun setFlying(flying: Boolean): IUser

    fun isFlying(): Boolean

    fun setVanished(vanished: Boolean): IUser

    fun isVanished(): Boolean

    fun getBase(): Player?

    fun kick(message: Component): Boolean

    fun checkCanFly()


}